server/app/com/xsn/explorer/util/RetryableFuture.scala

```diff
@@ -0,0 +1,59 @@
+package com.xsn.explorer.util
+
+import scala.concurrent.{ExecutionContext, Future}
+import akka.actor.Scheduler
+import akka.pattern.Patterns.after
+
+import scala.concurrent.duration._
+import scala.util.Try
+
+object RetryableFuture {
+
+  def apply[A](
+      delays: List[FiniteDuration]
+  )(
+      shouldRetry: Try[A] => Boolean
+  )(f: => Future[A])(implicit ec: ExecutionContext, scheduler: Scheduler): Future[A] = {
+    delays.foldLeft(f) {
+      case (result, delay) =>
+        result.transformWith { t =>
+          if (shouldRetry(t)) {
+            after(delay, scheduler, ec, Future.successful(1)).flatMap(_ => f)
+          } else {
+            result
+          }
+        }
+    }
+  }
+
+  def withExponentialBackoff[A](
+      initialDelay: FiniteDuration,
+      maxDelay: FiniteDuration
+  )(implicit ec: ExecutionContext, scheduler: Scheduler) = {
+    def generateDelays: (Long, Long, Int) => List[Long] = (initialDelay: Long, maxDelay: Long, retry: Int) => {
+      val currentDelay = getDelay(initialDelay, retry).toMillis
+      if (currentDelay > maxDelay) {
+        List.empty[Long]
+      } else {
+        currentDelay :: generateDelays(initialDelay, maxDelay, retry + 1)
+      }
+    }
+
+    val delays = generateDelays(initialDelay.toMillis, maxDelay.toMillis, 0).map(_.millis)
+    RetryableFuture[A](delays) _
+  }
+
+  def withExponentialBackoff[A](
+      initialDelay: FiniteDuration,
+      maxRetries: Int
+  )(implicit ec: ExecutionContext, scheduler: Scheduler) = {
+    val initialDelayMillis = initialDelay.toMillis
+    val delays = List.tabulate(maxRetries)(n => getDelay(initialDelayMillis, n))
+
+    RetryableFuture[A](delays) _
+  }
+
+  private def getDelay(baseDelay: Long, retry: Int, factor: Int = 2): FiniteDuration = {
+    (Math.pow(factor, retry) * baseDelay).longValue.millis
```

Comment:
Add a small random jitter when generating delays, `Random.nextInt(100)` should be enough.